src/clients/c++/request_http.cc

```diff
@@ -1127,8 +1127,9 @@ HttpRequestImpl::SetNextRawResult(
       }
     }
 
+    // std::cerr << "Output size - ob: " << size << " - " << ob << '\n';
     // If output couldn't accept any more bytes then move to the next.
-    if (ob == 0) {
+    if (ob == 0 || io->UsesSharedMemory()) {
```

Comment:
This seems like a strange place to put this check. Do you find that 'ob' !=0 for a shared-memory result? That would mean that some raw bytes were read for the result, and that shouldn't happen.

Reply:
The result size is used for extracting the request header. For the shared memory case although the result has a byte size the same is not used in calculation of the header. Moving this check to SetNextRawResult()